### ISSUE TITLE
Snowflake doesn't allow revoke OWNERSHIP anymore

### ISSUE THREAD

**Issue by lgwacker** [chanzuckerberg/terraform-provider-snowflake#304]
Hello guys,

Since last Snowflake update, revoke ownership is not available anymore. Only transfering ownership from one user/role to another.
Now, resource *snowflake_schema_grant* won't work if we try to change ownership of a schema, since it will try to revoke it first.

Guess we can change revoke code to, instead, grant ownership to the role terraform is using. But i don't know how to get that.

grant.go
```
// Revoke returns the SQL that will revoke privileges on the grant from the grantee
func (ge *CurrentGrantExecutable) Revoke(p string) string {
    if p == `OWNERSHIP` {
        return fmt.Sprintf(`GRANT %v ON %v %v TO ROLE "ACCOUNTADMIN" COPY CURRENT GRANTS`,
        p, ge.grantType, ge.grantName)
    } else {
        return fmt.Sprintf(`REVOKE %v ON %v %v FROM %v "%v"`,
            p, ge.grantType, ge.grantName, ge.granteeType, ge.granteeName)
    }
}
```

I added this if p == `OWNERSHIP` to explicit what could be done. If instead of hardcoding ACCOUNTADMIN , we could retrieve terraform's role, problem should be solved.

**Comment by a129722** [comment]
A stop-gap solution could be the following:

```
func (ge *CurrentGrantExecutable) Revoke(p string) string {
	if p == `OWNERSHIP` {
		currentRole := os.Getenv(`SNOWFLAKE_ROLE`)

		if currentRole != "" {
			return fmt.Sprintf(`GRANT OWNERSHIP ON %v %v TO ROLE "%v" COPY CURRENT GRANTS`,
				ge.grantType, ge.grantName, currentRole)
		}
	}
	return fmt.Sprintf(`REVOKE %v ON %v %v FROM %v "%v"`,
		p, ge.grantType, ge.grantName, ge.granteeType, ge.granteeName)
}
```

here we obtain the current role from the `SNOWFLAKE_ROLE` environment variable.  That's the only place where the current role is accessible at this point.  However, this doesn't work for cases where the role is supplied through terraform config or if no role is supplied and the default role for the user is utilized.

Alternatively within the `ConfigureProvider` method of `pkg/provider/provider.go`, we can issue the query of `SELECT CURRENT_ROLE()` and use the result elsewhere.  The cleanest way of threading things through would be to utilize the `meta` value for the provider.  That would require changes all over the code as there's an assumption that the `meta` field is of type `sql.DB`.

Any thoughts on this?

**Comment by lgwacker** [comment]
Amazing @a129722 ! So we have quite some ways to fix this problem. @ryanking @edulop91 Do you guys have any thoughts on the best way to proceed?

**Comment by ryanking** [comment]
cc also @alldoami

Can you clarify for me - is this something that snowflake changed? or that we changed in the provider?

**Comment by lgwacker** [comment]
@ryanking In october update, Snowflake stopped alowing revoke of ownership.
From their documentation:

> Transfers ownership of an object (or all objects of a specified type in a schema) from one role to another role. OWNERSHIP is a special type of privilege that can only be granted from one role to another role; it cannot be revoked. For more details, see Access Control in Snowflake.

**Comment by ryanking** [comment]
Ok thanks. I am currently thinking that we set to [`CURRENT_ROLE()`](https://docs.snowflake.com/en/sql-reference/functions/current_role.html). Thoughts?

**Comment by a129722** [comment]
Setting to `CURRENT_ROLE()` makes sense, but we do need to obtain what the value is and supply it to the `Revoke` method.  Unfortunately the statement of `GRANT OWNERSHIP ON TABLE MY_TABLE TO ROLE CURRENT_ROLE()` does not work.

**Comment by ryanking** [comment]
This should be fixed by #360.
